src/main/java/com/alexfu/sqlitequerybuilder/utils/Preconditions.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.alexfu.sqlitequerybuilder.utils;
+
+public final class Preconditions {
+  private Preconditions() {
+    // No op
+  }
+
+  /**
+   * Ensures that an object reference passed as a parameter to the calling method is not null.
+   *
+   * @param reference an object reference
+   * @return the non-null reference that was validated
+   * @throws NullPointerException if {@code reference} is null
+   */
+  public static <T> T checkNotNull(T reference) {
+    if (reference == null) {
+      throw new NullPointerException();
+    }
+
+    return reference;
+  }
+
+  /**
+   * Ensures that an object reference passed as a parameter to the calling method is not null.
+   *
+   * @param reference an object reference
+   * @param message exception message
+   * @return the non-null reference that was validated
+   * @throws NullPointerException if {@code reference} is null
+   */
+  public static <T> T checkNotNull(T reference, String message) {
+    if (reference == null) {
+      throw new NullPointerException(message);
+    }
+
+    return reference;
+  }
+
+  public static String checkNotEmpty(String name, String message) {
+    checkNotNull(name, message);
```

Comment:
I understand that for now the only use case for `checkArgument` is to check if a `String` argument is not empty, but there probably will be more cases as more features are added.

> Although I do think checkArgument could end up just replacing the existing methods we have in Preconditions

As mentioned above, the main reason to have separate methods is to throw clear exceptions. For e.g.

`Preconditions.checkArgument(value != null, "Value cannot be null")`

Altought the message tells you what's wrong, the method throws an `IllegalArgumentException` which might cause confusion.